mlflow/pyspark/optuna/study.py

```diff
@@ -0,0 +1,188 @@
+        :caption: Example
+
+        from mlflow.optuna.storage import MlflowStorage
+        from mlflow.pyspark.optuna.study import MlflowSparkStudy
+
+
+        def objective(trial):
+            x = trial.suggest_float("x", -10, 10)
+            return (x - 2) ** 2
+
+
+        experiment_id = "507151065975140"
+        study_name = "spark_mlflow_storage"
+
+        storage = MLFlowStorage(experiment_id=experiment_id)
+        mlflow_study = MlflowSparkStudy(
+            study_name, storage, mlflow_tracking_uri=mlflow.get_tracking_uri()
+        )
+        mlflow_study.optimize(objective, n_trials=4)
+    """
+
+    def __init__(
+        self,
+        study_name: str,
+        storage: MlflowStorage,
+        sampler: Optional["samplers.BaseSampler"] = None,
+        pruner: Optional[pruners.BasePruner] = None,
+        mlflow_tracking_uri: Optional[str] = "databricks",
+    ):
+        self.study_name = study_name
+        self._storage = storages.get_storage(storage)
+        self.sampler = sampler or samplers.TPESampler()
+        self.pruner = pruner or pruners.MedianPruner()
+
+        self.spark = SparkSession.active()
+
+        # check whether the SparkConnect mode
+        self._is_spark_connect_mode = is_spark_connect_mode()
+        self._mlflow_tracking_env = mlflow_tracking_uri
+
+        mlflow.set_tracking_uri(self._mlflow_tracking_env)
+        self._study = optuna.create_study(
+            study_name=self.study_name, sampler=self.sampler, storage=self._storage
+        )
+        self._study_id = storage.get_study_id_from_name(self.study_name)
+        self._directions = self._storage.get_study_directions(self._study_id)
+
+        if not isinstance(self._storage, MlflowStorage):
+            raise ValueError(
+                f"MlflowSparkStudy only works with `MlflowStorage`. But get {type(self._storage)}."
+            )
+
+    def optimize(
+        self,
+        func: "optuna.study.study.ObjectiveFuncType",
+        n_trials: Optional[int] = None,
+        timeout: Optional[float] = None,
+        n_jobs: int = 1,
+        catch: Iterable[type[Exception]] = (),
+        callbacks: Optional[Iterable[Callable[[Study, FrozenTrial], None]]] = None,
+    ) -> None:
+        experiment_id = self._storage._experiment_id
+        study_name = self.study_name
+        mlflow_tracking_env = self._mlflow_tracking_env
+        sampler = self.sampler
+
+        def run_task_on_executor_pd(iterator):
+            import traceback
+
+            import optuna
+            import pandas as pd
+
+            import mlflow
+            from mlflow.optuna.storage import MlflowStorage
+
+            mlflow.set_tracking_uri(mlflow_tracking_env)
+
+            storage = MlflowStorage(experiment_id=experiment_id)
+            study = optuna.load_study(study_name=study_name, sampler=sampler, storage=storage)
+            num_trials = sum(map(len, iterator))
+
+            try:
+                error_message = []
```

Comment:
```suggestion
                error_messages = []
```


`error_message` is misleading. It's a list. Should be plural.